### PR TITLE
feat: provide gpg identity signature check

### PR DESCRIPTION
**Pull request by smira**
Breaking change: `.conform.yaml` `gpg: bool` field was changed to be a
structure:

```yaml
gpg:
  required: true
  identity:
    gitHubOrganization: talos-systems
```

(`identity` is not required)

This enforces that GPG signature should come from one the members of the
GitHub organization to succeed, i.e. commit is signed by the member of
the organization.

Signed-off-by: Andrey Smirnov <andrey.smirnov@talos-systems.com>

### CODE REVIEW

File: internal/git/git.go [hunk 1]
```diff
@@ -10,12 +10,14 @@ import (
...
 	"path"
 	"path/filepath"
+	"strings"
 
 	git "github.com/go-git/go-git/v5"
 	"github.com/go-git/go-git/v5/config"
 	"github.com/go-git/go-git/v5/plumbing"
 	"github.com/go-git/go-git/v5/plumbing/object"
 	"github.com/go-git/go-git/v5/plumbing/storer"
+	"github.com/keybase/go-crypto/openpgp"
```
Comment (smira) [inline internal/git/git.go:20]: using keybase go-crypto fork for openpgp, as it supports modern keys

File: internal/policy/commit/commit.go [hunk 2]
```diff
@@ -57,6 +57,9 @@ type Commit struct {
 	DCO bool `mapstructure:"dco"`
 	// GPG enables the GPG signature check.
 	GPG bool `mapstructure:"gpg"`
+	// GPGSignatureGitHubOrganization enforces that GPG signature should come from
+	// one of the members of the GitHub org.
+	GPGSignatureGitHubOrganization string `mapstructure:"gpgSignatureGitHubOrg"`
```
Comment (smira) [inline internal/policy/commit/commit.go:62]: doesn't look short to me, but I don't have any better options? 

should it be subkey under `gpg`?

File: internal/policy/commit/commit.go [hunk 3]
```diff
@@ -158,11 +160,11 @@ func (c Commit) firstWord() (string, error) {
 
 	if c.Conventional != nil {
 		groups = parseHeader(c.msg)
-		if len(groups) != 6 {
+		if len(groups) != 7 {
```
Comment (smira) [inline internal/policy/commit/commit.go:163]: this was a bug in `master` with conventional commits, as regexp got changed in some other PR

Comment (smira) [reply internal/policy/commit/commit.go:163]: looks like it got fixed in #204, but never merged (?)

File: internal/policy/commit/commit.go (hunk 2 above)
Comment (andrewrynhard) [reply internal/policy/commit/commit.go:62]: I personally would like `gpg` do be more than just a `bool`. Will be a good place to put all GPG related things. But I know that can/will be a PITA. I could go either way.

Comment (smira) [reply internal/policy/commit/commit.go:64]: I guess we can't make it backwards compatible?

Comment (andrewrynhard) [reply internal/policy/commit/commit.go:64]: No, but that hasn't stopped us in the past. There is a reason this project is still alpha 😄

Comment (smira) [reply internal/policy/commit/commit.go:64]: I don't see a way to change `gpg: true` to `gpg:\n  enabled: true` without breaking backwards compat or doing two structs and decoding into both of them...

Comment (andrewrynhard) [reply internal/policy/commit/commit.go:64]: Yeah, what I mean it I am not against a breaking change. Conform is still alpha for that reason.

Comment (smira) [reply internal/policy/commit/commit.go:64]: got you, I will make a change then

**Comment by smira** [comment]
```
POLICY         CHECK                        STATUS        MESSAGE                                                              
commit         Header Length                PASS          Header is 40 characters                                              
commit         Imperative Mood              PASS          Commit begins with imperative verb                                   
commit         Header Case                  PASS          Header case is valid                                                 
commit         Header Last Character        PASS          Header last character is valid                                       
commit         DCO                          PASS          Developer Certificate of Origin was found                            
commit         GPG                          PASS          GPG signature found                                                  
commit         GPG Identity                 PASS          Signed by "Andrey Smirnov <andrey.smirnov@talos-systems.com>"        
commit         Conventional Commit          PASS          Commit message is a valid conventional commit                        
commit         Spellcheck                   PASS          Commit contains 0 misspellings                                       
commit         Number of Commits            PASS          HEAD is 1 commit(s) ahead of refs/heads/master                       
commit         Commit Body                  PASS          Commit body is valid                                                 
license        File Header                  PASS          All files have a valid license header   
```

**Review by andrewrynhard — APPROVED** [review]

File: internal/policy/commit/check_gpg_identity.go [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+
+	return check
+}
+
+func getOrganizationMembers(ctx context.Context, organization string) ([]string, error) {
+	var members []struct {
+		Login string `json:"login"`
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, fmt.Sprintf("https://api.github.com/orgs/%s/members", organization), nil)
```
Comment (AlekSi) [inline internal/policy/commit/check_gpg_identity.go:90]: We already use github.com/google/go-github module, so why not use it there?

Comment (smira) [reply internal/policy/commit/check_gpg_identity.go:90]: :shrug: probably it's easier? but I can look into that (if that is even available)

Comment (smira) [reply internal/policy/commit/check_gpg_identity.go]: got it updated

File: internal/git/git.go [hunk 5]
```diff
@@ -101,6 +103,49 @@ func (g *Git) HasGPGSignature() (ok bool, err error) {
 	return ok, err
 }
 
+// VerifyPGPSignature validates PGP signature against a keyring.
+func (g *Git) VerifyPGPSignature(armoredKeyrings []string) (entity *openpgp.Entity, err error) {
```
Comment (AlekSi) [inline internal/git/git.go:107]: Named return parameters are not used?

**Review by AlekSi — APPROVED** [review]

**Pull request merged**